lib/eco/incparser/incparser.py

```diff
@@ -385,6 +400,9 @@ def pass2(self, node, offset, error_offset):
             offset += c.textlength()
 
     def is_retainable_subtree(self, node):
+        # Subtrees are currently always unretainable due to a potential bug in
+        # Wagners algorithm
```

Comment:
I can try. It's quite difficult to explain. I do have a test for it. I'm gonna try and add a more detailed explanation to the test and the reference the test here.